packages/react-native-reanimated/Common/cpp/reanimated/Fabric/ReanimatedCommitHook.cpp

```diff
@@ -28,6 +31,16 @@ RootShadowNode::Unshared ReanimatedCommitHook::shadowTreeWillCommit(
     ShadowTree const &,
     RootShadowNode::Shared const &,
     RootShadowNode::Unshared const &newRootShadowNode) noexcept {
+  auto surfaceId = newRootShadowNode->getSurfaceId();
+
+  {
+    auto lock = std::unique_lock<std::mutex>(mutex_);
+    if (!initializedSurfaces_.contains(surfaceId)) {
+      initializeLayoutAnimations_(surfaceId);
+      initializedSurfaces_.insert(surfaceId);
+    }
```

Comment:
Let's store max `SurfaceId` so far and enumerate all surfaces when max changes.

Reply:
🫡